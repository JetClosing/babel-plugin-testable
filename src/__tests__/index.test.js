import path from 'path';
import fs from 'fs';
import assert from 'assert';
import { transformFileSync } from 'babel-core';
import plugin from '..';

function trim(str) {
  return str.replace(/^\s+|\s+$/, '');
}

const fixturesDir = path.join(__dirname, '../__fixtures__');
fs.readdirSync(fixturesDir).forEach((caseName) => {
  it(`${caseName.split('-').join(' ')}`, () => {
    const fixtureDir = path.join(fixturesDir, caseName);
    const codePath = path.join(fixtureDir, 'code.js');

    const hasBabelRc = fs.existsSync(path.join(fixtureDir, '.babelrc'));
    const actual = transformFileSync(codePath, { plugins: [plugin], babelrc: hasBabelRc }).code;

    const expected = fs.readFileSync(path.join(fixtureDir, 'output.js')).toString();

    assert.equal(trim(actual), trim(expected));
  });
});
