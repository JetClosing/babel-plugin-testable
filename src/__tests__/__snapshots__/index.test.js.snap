// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel env test 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
// @testable
var testableConst = exports.testableConst = 'testable-const';

var privateConst = 'private-const';"
`;

exports[`plugin disabled 1`] = `
"'use strict';

// @testable
var testableConst = 'testable-const';

var privateConst = 'private-const';"
`;
